src/game_logic/game.rs

```diff
@@ -401,6 +403,15 @@ impl Game {
         self.update_spawn_points();
     }
 
+    fn maybe_add_special_block_to_random_player(&self) {
+        match BlockType::from_score(self.score) {
+            BlockType::Normal => {}
+            special => {
+                self.players.choose(&mut rand::thread_rng()).unwrap().borrow_mut().next_block_queue.push_back(FallingBlock::new(special));
```

Comment:
Shouldn't push if the queue is already too long. Unlikely to happen, but could be used to DDOS.